src/States/MapViewStateDraw.cpp

```diff
@@ -69,49 +69,60 @@ void MapViewState::drawMiniMap()
 	bool isHeightmapToggled = mBtnToggleHeightmap.toggled();
 	renderer.drawImage(isHeightmapToggled ? mHeightMap : mMapDisplay, miniMapBoxFloat.startPoint());
 
-	if (ccLocationX() != 0 && ccLocationY() != 0)
+	const auto miniMapOffset = mMiniMapBoundingBox.startPoint() - NAS2D::Point<int>{0, 0};
+	const auto ccPosition = ccLocation();
+	if (ccPosition != NAS2D::Point<int>{0, 0})
 	{
-		renderer.drawSubImage(mUiIcons, ccLocationX() + mMiniMapBoundingBox.x() - 15, ccLocationY() + mMiniMapBoundingBox.y() - 15, 166, 226, 30, 30);
-		renderer.drawBoxFilled(ccLocationX() + mMiniMapBoundingBox.x() - 1, ccLocationY() + mMiniMapBoundingBox.y() - 1, 3, 3, 255, 255, 255);
+		const auto ccOffsetPosition = ccPosition + miniMapOffset;
+		const auto ccCommRangeImageRect = NAS2D::Rectangle<int>{166, 226, 30, 30};
+		renderer.drawSubImage(mUiIcons, ccOffsetPosition - ccCommRangeImageRect.size() / 2, ccCommRangeImageRect);
+		renderer.drawBoxFilled(NAS2D::Rectangle<int>::Create(ccOffsetPosition - NAS2D::Vector<int>{1, 1}, NAS2D::Vector<int>{3, 3}), NAS2D::Color::White);
 	}
 
-	for (auto _tower : Utility<StructureManager>::get().structureList(Structure::CLASS_COMM))
+	auto& structureManager = Utility<StructureManager>::get();
+	for (auto commTower : structureManager.structureList(Structure::CLASS_COMM))
 	{
-		if (_tower->operational())
+		if (commTower->operational())
 		{
-			Tile* t = Utility<StructureManager>::get().tileFromStructure(_tower);
-			renderer.drawSubImage(mUiIcons, t->x() + mMiniMapBoundingBox.x() - 10, t->y() + mMiniMapBoundingBox.y() - 10, 146, 236, 20, 20);
+			const auto commTowerPosition = structureManager.tileFromStructure(commTower)->position();
+			const auto commTowerRangeImageRect = NAS2D::Rectangle<int>{146, 236, 20, 20};
+			renderer.drawSubImage(mUiIcons, commTowerPosition + miniMapOffset - commTowerRangeImageRect.size() / 2, commTowerRangeImageRect);
 		}
 	}
 
-	for (auto _mine : mTileMap->mineLocations())
+	for (auto minePosition : mTileMap->mineLocations())
```

Comment:
This is an unnecessary copy of a variable that isn't changed. Use `const auto& minePosition`.

Reply:
Mmm, not sure. The `Point` class only has two fields, so a copy may be preferred to indirection. Plus, it seems quite likely the compiler could optimize away the copy.

Reply:
Well, at least mark it `const`.

Reply:
Came across some interesting stuff on the topic:
[Auto Type Deduction in Range-Based For Loops](https://blog.petrzemek.net/2016/08/17/auto-type-deduction-in-range-based-for-loops/)
StackOverflow: [C++ auto& vs auto](https://stackoverflow.com/questions/29859796/c-auto-vs-auto)

The note on copy elision was interesting:

> Use auto when you need a local copy. This will never produce a reference. The copy (or move) constructor must exist, but it might not get called, due to the copy elision optimization.


----

At any rate, I just noticed the use of `auto` was pre-existing, and not something I had changed here. It's perhaps a bit out of scope for this PR. Lacking a clear reason to change this, I'm going to leave it for now.